### ISSUE TITLE
`MergedAnnotations` search does not find container for repeatable annotation

### ISSUE THREAD

**Issue by heihei180** [spring-projects/spring-framework#32731]
when i use `applicationContext.getBeansWithAnnotation(Extensions.class)`;
in spring framework 5.3.20 , i can got a bean,
in spring framework 5.3.28, i can't got a bean.

---
definition of `Extensions`:
```java
@Inherited
@Retention(RetentionPolicy.RUNTIME)
@Target({ElementType.TYPE})
@Component
public @interface Extensions {

    String[] bizId() default BizScenario.DEFAULT_BIZ_ID;

    String[] useCase() default BizScenario.DEFAULT_USE_CASE;

    String[] scenario() default BizScenario.DEFAULT_SCENARIO;

    Extension[] value() default {};

}
``` 
Definition of variable "applicationContext":

```java
@Component
public class ExtensionBootstrap implements ApplicationContextAware {
    private ApplicationContext applicationContext;

    @Override
    public void setApplicationContext(ApplicationContext applicationContext) throws BeansException {
        this.applicationContext = applicationContext;
    }
}
```

Can someone tell me what changes have occurred here?
I think this should be a problem, but I couldn't find where the problem occurred.

**Comment by snicoll** [comment]
@heihei180  I am not sure what could have changed this. The code in text you've shared isn't actionable unfortunately, please share a small sample we can run ourselves that reproduces this behavior. You can attach a zip here or push the code to a GitHub repository.

**Comment by snicoll** [comment]
Thanks for the sample. I can reproduce, including with the latest release and I don't know if that's intended. 

Relying on the container annotation for this looks a bit odd to me though. If you use `Extension` it works obviously and I would expect you to use that, rather than the `@Repeatable` container. I've asked @sbrannen if that rings a bell.

**Comment by sbrannen** [comment]
Thanks for the demo projects, @heihei180. 👍 

I've reduced this to the following test.

```java
@Test
void test() {
	Extensions extensions = FooService.class.getAnnotation(Extensions.class);
	assertThat(extensions).as("@Extensions").isNotNull();

	extensions = MergedAnnotations.from(FooService.class, SearchStrategy.TYPE_HIERARCHY)
			.get(Extensions.class)
			.synthesize(MergedAnnotation::isPresent)
			.orElse(null);
	assertThat(extensions).as("@Extensions").isNotNull();
}
```

`java.lang.Class.getAnnotation(Class<Extensions>)` of course finds `@Extensions` on `FooService`, but Spring's `MergedAnnotations` support no longer does.

I consider that a regression, and I'll investigate further.

**Comment by sbrannen** [comment]
The above test actually fails on `5.3.20` as well if you remove all attributes from `Extensions.class` except the `value` attribute.

This appears to be related to #29685, and I assume this bug may have existed since Spring Framework 5.2 when the `MergedAnnotations` infrastructure was introduced.

**Comment by snicoll** [comment]
> I would like to confirm, will the Spring team support in the future?

Perhaps you're looking at this issue on your phone? On a desktop you can see on the right hand side that it's flagged as a regression, scheduled for `6.1.7` and for backports. If you follow the link above you can see backports for `6.0.x` and `5.3.x` are scheduled. So, yes, the plan is to address this issue.

**Comment by sbrannen** [comment]
This has been addressed in 4baad16437524f6f16f61fcfb3dc0458f7aaff47 which will be included in Spring Framework `6.1.7`, `6.0.20`, and `5.3.35`.
